src/tables/sortable-table.js

```diff
@@ -36,37 +36,89 @@ import classnames from 'classnames'
 /* eslint react/jsx-key: 0 */
 
 const propTypes = {
+  className: PropTypes.string,
   columns: PropTypes.arrayOf(Types.column).isRequired,
-  controlled: PropTypes.bool.isRequired,
   data: PropTypes.arrayOf(PropTypes.object).isRequired,
+  initialSortPath: PropTypes.string.isRequired,
+  initialSortFunc: PropTypes.func,
+  initialAscending: PropTypes.bool,
+  disableReverse: PropTypes.bool,
   disableSort: PropTypes.bool.isRequired,
+  controlled: PropTypes.bool,
+  onChange: PropTypes.func,
   rowComponent: Types.component,
   headerComponent: Types.component,
-  className: PropTypes.string,
-  ...sortablePropTypes,
 }
-
 const defaultProps = {
-  className: ''
+  className: '',
+  initialAscending: true,
+  disableReverse: false,
+  controlled: false,
+  onChange: noop,
 }
 
-function SortableTable ({
+
+function SortableTable({
+  className,
   columns,
   data: unsortedData,
+  initialSortPath,
+  initialSortFunc,
+  initialAscending,
+  disableReverse,
   disableSort,
   controlled,
-  sort,
-  ascending,
-  sortPath,
-  setSortPath,
-  setSortFunc,
+  onChange,
   rowComponent,
   headerComponent,
-  className,
 }) {
-  const data = (controlled || disableSort) ? unsortedData : sort(unsortedData)
+  const [sortControls, setSortControls] = useState({
+    ascending: initialAscending,
+    sortPath: initialSortPath,
+    sortFunc: initialSortFunc
+  })
+  const [data, setData] = useState(unsortedData)
+  const { ascending, sortPath, sortFunc } = sortControls
+
+  const handleColumnChange = (column) => {
+    if (column.disabled) return
+
+    const newSortPath = column.name
+    const newSortFunc = column.sortFunc || null
+
+    // Toggle ascending if the path is already selected. Otherwise, default
+    // to ascending when switching paths...
+    const newAscending = newSortPath === sortPath ? !ascending : true
+
+    setSortControls({
+      ...sortControls,
+      ascending: newAscending,
+      sortPath: newSortPath,
+      sortFunc: newSortFunc,
+    })
+  }
+
+  useEffect(() => {
```

Comment:
Instead of using `setData` declaratively, you can derive the sorted data using `useMemo`.

```
const sortedData = useMemo(() => {...}, [ ascending, sortPath, sortFunc ])
```

And onChange can be triggered within handleColumnChange.

Reply:
Done.